backend/api/projects.py

```diff
@@ -83,26 +84,36 @@ def add_projects():
 @projects.route('/api/projects', methods=['GET'])
 @login_required
 def get_projects():
-    """Return list of projects, args->limit, skip
+    """Return list of projects, args->limit, skip, archived
 
     Returns:
         res: A list of projects
     """
     limit = request.args.get('limit', type=int)
     skip = request.args.get('skip', type=int)
+    archived = request.args.get('archived', type=str, default='false')
+    print(request.query_string, file=sys.stderr)
+
+    print(archived, file=sys.stderr)
     argc = len(request.args)
 
     if g.projects.count() == 0:
         return make_response(jsonify([]), 200)
-
-    if argc == 0:
-        res = g.projects.find({})
-    elif limit and skip and argc < 3:
-        res = g.projects.find({}, limit=limit, skip=skip)
-    elif limit and argc < 2:
-        res = g.projects.find({}, limit=limit)
-    elif skip and argc < 2:
-        res = g.projects.find({}, skip=skip)
+    query = {}
+    if archived == 'true':
+        query = {'archived': True}
+    elif archived == 'false':
+        query = {'archived': False}
+    elif archived == 'mixed':
+        query = {}
+    if limit and skip:
+        res = g.projects.find(query, limit=limit, skip=skip)
+    elif limit:
+        res = g.projects.find(query, limit=limit)
+    elif skip:
+        res = g.projects.find(query, skip=skip)
+    elif archived:
+        res = g.projects.find(query)
```

Comment:
moved to checking for an allowed status for archived because i's a string to support mixed as an argument